src/components/DownloadImageDialog/ImageForm.tsx

```diff
@@ -204,347 +251,550 @@ export const ImageForm = memo(function ImageForm({
+						expandIcon={<FontAwesomeIcon icon={faChevronRight} />}
+						sx={{ flexDirection: 'row-reverse', gap: 1 }}
+					>
+						<Typography variant="titleSm">Advanced settings</Typography>
+					</AccordionSummary>
+					<AccordionDetails>
+						<Stack>
+							<FormControl>
+								<FormLabel
+									htmlFor="poll-interval-label"
+									sx={{ display: 'flex', alignItems: 'center' }}
+								>
+									Check for updates every X minutes{' '}
+									<MUILinkWithTracking
+										eventProperties={{
+											source:
+												'Application Add Device Modal Poll Interval Doc Icon',
+										}}
+										href={POLL_INTERVAL_DOCS}
+										sx={{
+											display: 'flex',
+											alignItems: 'center',
+											height: '1.5rem',
+										}}
+									>
+										<ArticleIcon sx={{ ml: 1, fontSize: '1.15rem' }} />
+									</MUILinkWithTracking>
+								</FormLabel>
+								<TextField
+									id="poll-interval-label"
+									aria-labelledby="poll-interval-label"
+									value={model.appUpdatePollInterval}
+									slotProps={{
+										htmlInput: {
+											name: 'appUpdatePollInterval',
+											autoComplete: 'appUpdatePollInterval-auto-complete',
+										},
+									}}
+									onChange={(event) => {
+										onChange('appUpdatePollInterval', event.target.value);
+									}}
+								/>
+							</FormControl>
+							<InputLabel htmlFor="provision-key-name" sx={{ my: 2 }}>
+								Provisioning Key name
+							</InputLabel>
+							<TextField
+								name="provisioningKeyName"
+								id="provision-key-name"
+								value={model.provisioningKeyName ?? ''}
+								slotProps={{
+									htmlInput: {
+										name: 'provisioningKeyName',
+										autoComplete: 'provisioningKeyName-auto-complete',
+									},
+								}}
+								onChange={(event) => {
+									onChange('provisioningKeyName', event.target.value);
+								}}
+							/>
+							<InputLabel htmlFor="provision-key-expiring" sx={{ my: 2 }}>
+								Provisioning Key expiring on
+							</InputLabel>
+							<TextField
+								type="date"
+								id="provision-key-expiring"
+								value={model.provisioningKeyExpiryDate ?? ''}
+								slotProps={{
+									htmlInput: {
+										name: 'provisioningKeyExpiryDate',
+										autoComplete: 'provisioningKeyExpiryDate-auto-complete',
+									},
+								}}
+								onChange={(event) => {
+									onChange('provisioningKeyExpiryDate', event.target.value);
+								}}
+							/>
+						</Stack>
+					</AccordionDetails>
+				</Accordion>
 			</Box>
-			<Divider variant="fullWidth" sx={{ my: 3, borderStyle: 'dashed' }} />
-			<Button
-				onClick={() => {
-					setShowAdvancedSettings(!showAdvancedSettings);
+			<Dialog
+				open={showSecureBootConfirmationDialog}
+				onClose={() => {
+					setShowSecureBootConfirmationDialog(false);
+					dismissSecureBootWarning(false, dontShowSecureBootWarningAgain);
+					setDontShowSecureBootWarningAgain(false);
 				}}
```

Comment:
what about moving setShowSecureBootConfirmationDialog and setDontShowSecureBootWarningAgain inside dismissSecureBootWarning ? 
Why do we need to reset `dontShowSecureBootWarningAgain` to false ? if I reopen the modal I should see what I've selected right ?